napari/_tests/test_magicgui.py

```diff
@@ -61,6 +63,78 @@ def add_data() -> dtype:  # type: ignore
     assert viewer.layers[0].source.widget == add_data
 
 
+def test_magicgui_add_data_inheritance(make_napari_viewer):
+    viewer = make_napari_viewer()
+    viewer.add_image(np.random.rand(10, 10), scale=(2, 2), translate=(1, 1))
+
+    @magicgui
+    def add_data(data: types.ImageData) -> types.LabelsData:
+        return (data > 0.5).astype('uint8')
+
+    viewer.window.add_dock_widget(add_data)
+    add_data()
+    assert len(viewer.layers) == 2
+    assert isinstance(viewer.layers[1], Labels)
+    npt.assert_array_equal(viewer.layers[1].scale, (2, 2))
+    npt.assert_array_equal(viewer.layers[1].translate, (1, 1))
+
+
+def test_magicgui_add_layer_inheritance(make_napari_viewer):
+    viewer = make_napari_viewer()
+    viewer.add_image(np.random.rand(10, 10), scale=(2, 2), translate=(1, 1))
+
+    class SampleEnum(Enum):
+        A = 'A'
+        B = 'B'
+
+    @magicgui
+    def add_data(
+        data: Image, factor: float = 0.5, combo: SampleEnum = SampleEnum.A
```

Comment:
Can you expand a little on how this test is different to the others? Is it just that the function is requesting a `Layer` rather than a `LayerData`? If so, what is the purpose of the additional combo? Is it to make sure that it's ignored because it's not requesting a layer? Maybe we can add a docstring to this test with a quick explanation.